### ISSUE TITLE
Favicon loading inconsistently and/or not updating

### ISSUE THREAD

**Issue by Danziger** [plausible/analytics#2005]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Plausible Cloud from plausible.io

### Describe the bug

There are two separated but related issues:

1. Favicons in subdirectories do not load. For example, in this sit: https://danziger.github.io/stylophonejs/.

2. Favicons in the site selector (current value, loaded from `https://icons.duckduckgo.com/ip3/...`) do not match the ones inside the selector's options (loaded from Plausible's own server).

### Expected behavior

1. Favicons should always load properly.
2. The favicon for a given site should be shown consistently across the page.

### Screenshots

![image](https://user-images.githubusercontent.com/6564894/178303787-12acf10a-e0a1-4b64-8c98-1017db8469e8.png)

### Environment

```markdown
- OS: Windows 10
- Browser: Chrome
- Browser Version: `Version 103.0.5060.114 (Official Build) (64-bit)`
```

**Comment by Danziger** [comment]
@metmarkosaric Well, in the screenshot above, the icon next to the current select value points to https://icons.duckduckgo.com/ip3/solman.cat.ico while the one in the list (for the same site) points to https://plausible.io/favicon/sources/solman.cat, which seems to be rendering the right image.

If you were using your internal URL instead of directly pointing to DuckDuckGo, the second issue would be resolved.

For the first one, ok, that doesn't work in DuckDuckGo either... 😢

**Comment by metmarkosaric** [comment]
For me both of those links show the same favicon. Perhaps it was cache or something like that

**Comment by Danziger** [comment]
Yep, I can confirm issue number 2 is a caching issue. I tried to add a random query string to the favicon (`https://solman.cat/favicon.ico?v=f4c299d2-694d-4074-a50f-945064da35d4`) but that doesn't fix the issue either...

However, when I access the page from a different location (using a VPN) it looks fine.

I'll see if I can resolve issue number 1 using absolute URLs instead.

**Comment by metmarkosaric** [comment]
Thanks for the update @Danziger! I'll close this for now as everything seem to work the way it is intended.

**Comment by ronilaukkarinen** [comment]
I have similar issue. No favicon is showing up here (latest version), I rembember seeing them there a while back:

![image](https://github.com/plausible/analytics/assets/1534150/4b87521c-51e1-4fb2-8c1e-6a803187f743)

But most of them (not all) are showing up here:

![image](https://github.com/plausible/analytics/assets/1534150/b07f5ad6-d4d2-4008-ab2e-3142549904ad)

How to resolve? @metmarkosaric
